blobstore/src/test/java/org/jclouds/blobstore/integration/internal/BaseBlobIntegrationTest.java

```diff
@@ -799,7 +799,9 @@ public void testMetadata() throws InterruptedException, IOException {
          blob.getMetadata().getUserMetadata().put("Adrian", "wonderpuff");
          blob.getMetadata().getUserMetadata().put("Adrian", "powderpuff");
 
+         awaitConsistency();
          addBlobToContainer(container, blob);
+         awaitConsistency();
```

Comment:
You can change removeBlob to be strongly consistent - but I didn't want to change the behavior (as it would increase the scope of the PR).

The way to do that would be to abstract deletion better and use that in removeBlob as well as other places (such as cleaning up the empty directory tree).